integration/src/test/resources/org/mule/test/construct/sub-flow.xml

```diff
@@ -109,9 +107,7 @@
         <append-string-transformer message="x"/>
         <append-string-transformer message="y"/>
         <append-string-transformer message="z"/>
-        <component>
-            <singleton-object class="org.mule.tck.core.lifecycle.LifecycleTrackerCheckComponent"/>
-        </component>
+        <custom-processor class="org.mule.tck.core.lifecycle.LifecycleTrackerCheckProcessor"/>
```

Comment:
The callback on the `test:processor` doesn't allow to change the payload, and this processor does update the payload. I migrated the cases where the payload is not modified or it is not needed by the test afterwards to use `test:component`, but that approach doesn't work for cases like this.